src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,78 @@
+package lotto.controller;
+
+import static lotto.util.Parser.parseLottoNumber;
+import static lotto.util.Parser.parseLottoNumberInteger;
+import static lotto.util.Parser.parsePriceInteger;
+
+import java.util.function.Supplier;
+import lotto.dto.LottoStatisticsDTO;
+import lotto.dto.LottosDTO;
+import lotto.model.BonusNumber;
+import lotto.model.Lotto;
+import lotto.model.LottoResult;
+import lotto.model.Lottos;
+import lotto.model.Price;
+import lotto.model.WinningLotto;
+import lotto.service.LottoService;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController {
+
+    private final InputView inputView = new InputView();
+    private final OutputView outputView = new OutputView();
+    private final LottoService lottoService = new LottoService();
+
+    public void run() {
+        Lottos lottos = lottoService.buyLottos(inputPrice());
+        printBoughtLottoInfo(LottosDTO.from(lottos));
+        WinningLotto winningLotto = inputWinningLotto();
+        LottoResult lottoStatistics = lottoService.getStatistics(winningLotto, lottos);
+        outputView.printStatistics(LottoStatisticsDTO.from(lottoStatistics, lottos.amount()));
+    }
+
+    private WinningLotto inputWinningLotto() {
+        Lotto lotto = inputDrawLotto();
+        return handleInput(() -> lottoService.generateWinningLotto(
+                lotto,
+                inputBonusNumber()
+        ));
+    }
+
+    private BonusNumber inputBonusNumber() {
+        return handleInput(() -> {
+            outputView.printInputBonusNumber();
+            String bonusNumber = inputView.readBonusNumber();
+            return lottoService.generateBonusNumber(parseLottoNumberInteger(bonusNumber));
+        });
+    }
+
+    private Lotto inputDrawLotto() {
+        return handleInput(() -> {
+            outputView.printInputWinningNumbers();
+            String winningNumber = inputView.readWinningLottoNumber();
+            return lottoService.generateLotto(parseLottoNumber(winningNumber));
```

Comment:
이 부분을 인라인 하면 코드가 너무 길어지기 때문이 일부러 안 하신거 같아요 !

Reply:
맞습니다! 한줄에 넣으면 너무 길어지더라구요..
이런 형태로도 짤수 있을것같긴 하네요!
```suggestion
            return lottoService.generateLotto(parseLottoNumber(
                inputView.readWinningLottoNumber()
            ));
```